persim/images.py

```diff
@@ -210,3 +220,695 @@ def show(self, imgs, ax=None):
+        pers_dgm = np.copy(pers_dgm)
+
+        if skew:
+            pers_dgm[:, 1] = pers_dgm[:, 1] - pers_dgm[:, 0]
+            ylabel = 'persistence'
+        else:
+            ylabel = 'death'
+
+        # setup plot range
+        plot_buff_frac = 0.05
+        bmin = np.min((np.min(pers_dgm[:, 0]), np.min(self._bpnts)))
+        bmax = np.max((np.max(pers_dgm[:, 0]), np.max(self._bpnts)))
+        b_plot_buff = (bmax - bmin) * plot_buff_frac
+        bmin -= b_plot_buff
+        bmax += b_plot_buff
+
+        pmin = np.min((np.min(pers_dgm[:, 1]), np.min(self._ppnts)))
+        pmax = np.max((np.max(pers_dgm[:, 1]), np.max(self._ppnts)))
+        p_plot_buff = (pmax - pmin) * plot_buff_frac
+        pmin -= p_plot_buff
+        pmax += p_plot_buff
+
+        ax = ax or plt.gca()
+        ax.set_xlim(bmin, bmax)
+        ax.set_ylim(pmin, pmax)
+
+        # compute reasonable line width for pixel overlay (initially 1/50th of the width of a pixel)
+        linewidth = (1/50 * self.pixel_size) * 72 * plt.gcf().bbox_inches.width * ax.get_position().width / \
+                    np.min((bmax - bmin, pmax - pmin))
+
+        # plot the persistence image grid
+        if skew:
+            hlines = np.column_stack(np.broadcast_arrays(self._bpnts[0], self._ppnts, self._bpnts[-1], self._ppnts))
+            vlines = np.column_stack(np.broadcast_arrays(self._bpnts, self._ppnts[0], self._bpnts, self._ppnts[-1]))
+            lines = np.concatenate([hlines, vlines]).reshape(-1, 2, 2)
+            line_collection = LineCollection(lines, color='black', linewidths=linewidth)
+            ax.add_collection(line_collection)       
+
+        # plot persistence diagram
+        ax.scatter(pers_dgm[:, 0], pers_dgm[:, 1])
+
+        # plot diagonal if necessary
+        if not skew:
+            min_diag = np.min((np.min(ax.get_xlim()), np.min(ax.get_ylim())))
+            max_diag = np.min((np.max(ax.get_xlim()), np.max(ax.get_ylim())))
+            ax.plot([min_diag, max_diag], [min_diag, max_diag])
+
+        # fix and label axes
+        ax.set_aspect('equal')
+        ax.set_xlabel('birth')
+        ax.set_ylabel(ylabel)
+
+        # optionally save figure
+        if out_file:
+            plt.savefig(out_file, bbox_inches='tight')
+
+    def plot_image(self, pers_img, ax=None, out_file=None):
+        """
+        Plot a persistence image
+        :param pers_img: (N,K) numpy array encoding a persistence image, e.g. output of PersistenceImager.transform()
+        :param out_file: optional path to save the persistence diagram 
+        """
+        ax = ax or plt.gca()
+        ax.matshow(pers_img.T, **{'origin': 'lower'})
+
+        # fix and label axes
+        ax.set_xlabel('birth')
+        ax.set_ylabel('persistence')
+        ax.get_xaxis().set_ticks([])
+        ax.get_yaxis().set_ticks([])
+
+        # optionally save figure
+        if out_file:
+            plt.savefig(out_file, bbox_inches='tight')
+
+def dict_print(dict_in):
```

Comment:
Could you use the prettyprint module instead for this? https://docs.python.org/3/library/pprint.html

Reply:
Done. I removed dict_print, and replaced where it was used with pprint.